plugins/catalog/src/components/EntityLayout/EntityLayout.tsx

```diff
@@ -245,17 +260,20 @@ export const EntityLayout = (props: EntityLayoutProps) => {
         pageTitleOverride={headerTitle}
         type={headerType}
       >
-        {entity && (
-          <>
-            <EntityLabels entity={entity} />
-            <EntityContextMenu
-              UNSTABLE_extraContextMenuItems={UNSTABLE_extraContextMenuItems}
-              UNSTABLE_contextMenuOptions={UNSTABLE_contextMenuOptions}
-              onUnregisterEntity={() => setConfirmationDialogOpen(true)}
-              onInspectEntity={() => setInspectionDialogOpen(true)}
-            />
-          </>
-        )}
+        {entity &&
+          (Boolean(actions) ? (
+            actions
+          ) : (
+            <>
+              <EntityLabels entity={entity} />
+              <EntityContextMenu
```

Comment:
This menu is very tied to this component and tricky to reimplement. Would it be enough for your use-case to just be able to replace the `EntityLabels` component instead and keep the menu intact? If not I'm feeling that the best way forward at this point is that you implement your own `EntityLayout` component with the customizations that you need

Reply:
I based the implementation on @freben comment here #16173 where he suggested taking the approach of all or nothing.

Reply:
Yep and we discussed it a bit more and came to the conclusion that removing the context menu wouldn't be very helpful, but that this would instead be something you can use if you want to customize the labels and add additional items to the left.